### PR TITLE
feat: cursor encode/decode

### PR DESCRIPTION
**Pull request by alan910127**

### CODE REVIEW

**Comment by alan910127** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/3?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/4?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
        * **PR #5** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/5?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #6** <a href="https://app.graphite.dev/github/pr/GymSquad/archive-api/6?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/4?utm_source=stack-comment).

File: app/websites/routes.py [hunk 1]
```diff
@@ -36,15 +37,120 @@ async def update_website(
...
+    if cursor is not None:
+        match SearchWebsiteCursor.from_base64(cursor):
+            case Ok(obj):
+                cursor_obj = obj
+            case Err(err):
+                logger.warning(
+                    "Invalid cursor received",
+                    cursor=cursor,
+                    error=err,
+                )
```
Comment (BaBa0525) [inline app/websites/routes.py:151]: 這個 log 會存在哪裡

**Review by BaBa0525 — APPROVED** [review]
Good

[beautiful](https://app.graphite.dev/macros/GymSquad)

**Comment by alan910127** [comment]
> 
> 
> 這個 log 會存在哪裡

現在還只會輸出到 console

**Comment by alan910127** [comment]
### Merge activity

* **Jan 27, 6:16 AM**: @alan910127 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/4).
* **Jan 27, 6:19 AM**: [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/4) rebased this pull request as part of a merge.
* **Jan 27, 6:20 AM**: @alan910127 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/GymSquad/archive-api/4).

**Pull request merged**